ocaml/database/block_device_io.ml

```diff
@@ -326,13 +326,22 @@ let accept_conn s latest_response_time =
   let now = Unix.gettimeofday () in
   let timeout = latest_response_time -. now in
   (* Await an incoming connection... *)
-  let ready_to_read, _, _ = Unix.select [s] [] [] timeout in
-  R.debug "Finished selecting" ;
-  if List.mem s ready_to_read then
-    (* We've received a connection. Accept it and return the socket. *)
-    fst (Unix.accept s)
-  else (* We must have timed out *)
+  let epoll = Polly.create () in
+  Polly.add epoll s Polly.Events.inp ;
+  let fds =
+    Fun.protect
+      ~finally:(fun () -> Polly.close epoll)
+      (fun () ->
+        Polly.wait_fold epoll 1
+          (int_of_float (timeout *. 1000.))
+          []
+          (fun _ fd _ fds -> fst (Unix.accept fd) :: fds)
+      )
+  in
+  if fds = [] then (* We must have timed out *)
     raise Unixext.Timeout
+  else (* There will only ever be a maximum of one fd *)
+    List.hd fds
```

Comment:
While List.hd is guarded here, it's preferrable to destructure the list to obtain values from it:
```suggestion
  match fds with
  | [] -> (* We must have timed out *)
      raise Unixext.Timeout
  | fd :: _ -> (* There will only ever be a maximum of one fd *)
      fd
```